packages/web-components/src/theme/set-theme.ts

```diff
@@ -42,13 +42,12 @@ export const setTheme = (theme: Theme) => {
     themeStyleTextMap.set(theme, `html{${tokenDeclarations.join('')}}`);
   }
 
+  // Update the CSSStyleSheet with the new theme
+  themeStyleSheet.replaceSync(themeStyleTextMap.get(theme)!);
```

Comment:
Should we do this to avoid replacing stylesheet in non-Firefox?

```ts
if (!SUPPORTS_REGISTER_PROPERTY) {
  themeStyleSheet.replaceSync(themeStyleTextMap.get(theme)!);
}
```

Reply:
Gating based only on registerProperty support actually breaks in non-Firefox, so we'd need...

```js
  if (!SUPPORTS_REGISTER_PROPERTY) {
    // for adding themes to older browsers (e.g. current firefox)
    themeStyleSheet.replaceSync(themeStyleTextMap.get(theme)!);
  }

  if (!document.adoptedStyleSheets.includes(themeStyleSheet)) {
    // for future theme updates
    document.adoptedStyleSheets.push(themeStyleSheet);
  } else {
    // for updating the theme
    themeStyleSheet.replaceSync(themeStyleTextMap.get(theme)!);
  }
  ```

I'm trying to imagine cleaner ways, but I think the most straightforward thing is to always update the stylesheet (like in this PR).

Reply:
Ah you are right. OK, yeah let's go with always setting the stylesheet then. It shouldn't have much impact on perf anyways.